File: game-utils/default-bots/python3/my_bot.py
import asyncio

from core.bot import Bot
from core.networking_client import connect


# The default Python 3 bot implementation that is not specific
# to any Planet Lia game but instead serves as a base for
# all game specific bot implementations.
class MyBot(Bot):

    # Called only once before the match starts. It holds the
    # data that you may need before the game starts.
    def setup(self, initial_data):
        # TODO
        # This is the default bot that is not configured
        # for any specific game.
        # See how to support a specific game:
        # - <TODO_LINK>
        pass

    # Called repeatedly while the match is generating. Each
    # time you receive the current match state and can use
    # response object to issue your commands.
    def update(self, state, response):
        # TODO
        # This is the default bot that is not configured
        # for any specific game.
        # See how to support a specific game:
        # - <TODO_LINK>
        pass


# Connects your bot to match generator, don't change it.
if __name__ == "__main__":
    asyncio.get_event_loop().run_until_complete(connect(MyBot()))

File: games/lia-1/bots/java/src/MyBot.java
import core.*;
import core.api.*;

/**
 * The default Java bot implementation that is not specific
 * to any Planet Lia game but instead serves as a base for
 * all game specific bot implementations.
 */
public class MyBot implements Bot {

    Constants constants;

    // Called only once before the match starts. It holds the
    // data that you may need before the game starts.
    @Override
    public void setup(InitialData data) {
        // Save constants for later use
        constants = data.constants;
    }

    // This method is called 10 times per game second and holds current
    // game state. Use Api object to call actions on your units.
    // - GameState reference: https://docs.liagame.com/api/#gamestate
    // - Api reference:       https://docs.liagame.com/api/#api-object
    @Override
    public void update(MatchState state, Response response) {
        // If you have enough resources to spawn a new warrior unit then spawn it.
        if (state.resources >= constants.WARRIOR_PRICE) {
            response.spawnUnit(UnitType.WARRIOR);
        }

        // We iterate through all of our units that are still alive.
        for (int i = 0; i < state.units.length; i++) {
            UnitData unit = state.units[i];

            // If the unit is not going anywhere, we send it
            // to a random valid location on the map.
            if (unit.navigationPath.length == 0) {

                // Generate new x and y until you get a position on the map
                // where there is no obstacle. Then move the unit there.
                while (true) {
                    int x = (int) (Math.random() * constants.MAP_WIDTH);
                    int y = (int) (Math.random() * constants.MAP_HEIGHT);

                    // Map is a 2D array of booleans. If map[x][y] equals false it means that
                    // at (x,y) there is no obstacle and we can safely move our unit there.
                    if (!constants.MAP[x][y]) {
                        response.navigationStart(unit.id, x, y);
                        break;
                    }
                }
            }

            // If the unit is a worker and it sees at least one resource
            // then make it go to the first resource to collect it.
            if (unit.type == UnitType.WORKER && unit.resourcesInView.length > 0) {
                ResourceInView resource = unit.resourcesInView[0];
                response.navigationStart(unit.id, resource.x, resource.y);
            }

            // If the unit is a warrior and it sees an opponent then start shooting
            if (unit.type == UnitType.WARRIOR && unit.opponentsInView.length > 0) {
                response.shoot(unit.id);
                response.saySomething(unit.id, "I see you!");
            }
        }
    }

    // Connects your bot to match generator, don't change it.
    public static void main(String[] args) throws Exception {
        NetworkingClient.connectNew(args, new MyBot());
    }
}
